microsoft/testsuites/vm_extensions/azsecpack.py

```diff
@@ -0,0 +1,287 @@
+            else:
+                posix_os.add_azure_core_repo()
+
+            if posix_os.is_package_in_repo(package):
+                posix_os.install_packages(package)
+                log.info("azure-mdsd is installed successfully")
+            else:
+                raise UnsupportedDistroException(
+                    node.os, f"The distro doesn't have {package} in its repo"
+                )
+        else:
+            log.info("azure-mdsd is already installed")
+
+    def _install_monitor_agent_extension(self, node: Node) -> None:
+        # Use the Azure Monitor Linux Agent extension to install the monitoring agent
+        # This test uses autoconfig, the settings should be {"GCS_AUTO_CONFIG":true}
+        settings = {"GCS_AUTO_CONFIG": True}
+        extension = node.features[AzureExtension]
+        result = extension.create_or_update(
+            name="AzureMonitorLinuxAgent",
+            publisher="Microsoft.Azure.Monitor",
+            type_="AzureMonitorLinuxAgent",
+            type_handler_version="1.0",
+            auto_upgrade_minor_version=True,
+            enable_automatic_upgrade=True,
+            settings=settings,
+        )
+        assert_that(result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+        # After Azure Monitor Linux Agent extension is installed, the package
+        # azuremonitoragent should be installed.
+        posix_os: Posix = cast(Posix, node.os)
+        assert_that(posix_os.package_exists("azuremonitoragent")).described_as(
+            "Expected the azuremonitoragent package to be installed"
+        ).is_equal_to(True)
+
+    def _install_security_agent_extension(self, node: Node) -> None:
+        # Install AzureSecurityLinuxAgent VM extension
+        azsec_settings = {
+            "enableGenevaUpload": True,
+            "enableAutoConfig": True,
+        }
+        extension = node.features[AzureExtension]
+        result = extension.create_or_update(
+            name="AzureSecurityLinuxAgent",
+            publisher="Microsoft.Azure.Security.Monitoring",
+            type_="AzureSecurityLinuxAgent",
+            type_handler_version="2.0",
+            auto_upgrade_minor_version=True,
+            enable_automatic_upgrade=True,
+            settings=azsec_settings,
+        )
+        assert_that(result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+        # Check azure-security, azsec-monitor, azsec-clamav, auoms are installed
+        posix_os: Posix = cast(Posix, node.os)
+        azsec_packages = ["azure-security", "azsec-monitor", "azsec-clamav", "auoms"]
+        result = True
+        loop_count = 0
+        while loop_count < 30:
+            for package in azsec_packages:
+                result = posix_os.package_exists(package) and result
+                if result is False:
+                    break
```

Comment:
Should it raise exception, if some package doesn't exist?

Reply:
Yes, it should raise an exception. Have added.

Reply:
Haven't seen it's changed.

Reply:
I added a log, and by the end of this function, I added an assert